api/gateway/v1alpha1/peering_types.go

```diff
@@ -103,6 +104,36 @@ func (p *Peering) Validate(_ context.Context, _ kclient.Reader) error {
 	if len(vpcs) != 2 {
 		return fmt.Errorf("peering must have exactly 2 VPCs, got %d", len(vpcs)) //nolint:goerr113
 	}
+	for name, vpc := range p.Spec.Peering {
+		if vpc == nil {
+			continue
+		}
+		for _, expose := range vpc.Expose {
+			if len(expose.IPs) < len(expose.As) {
```

Comment:
I think it's a bit restrictive b/c it'll be totally valid to have a pair of /24 in the `IPs` and a single /23 in the `As`, as it would be the same number of IPs, but then `Not` could make that calculation even more complicated. I think we should be comparing amount of IPs referenced in both sections to be the same, but we can skip it for now as statefull NAT will change it anyways

Reply:
That's my understanding for the stateless NAT as we need 1 to 1 mapping for all IPs that we expose, we can add a check later, thank you!